test/docfx.RegressionTest/RegressionTest.cs

```diff
@@ -164,19 +210,35 @@ private static bool Test(Options opts, string workingFolder, string remoteBranch
             return true;
         }
 
-        private static string EnsureTestData(Options opts, string workingFolder)
+        private static string EnsureTestData(Options opts, string workingFolder, int retryCount = 0)
         {
-            var remoteBranch = string.IsNullOrEmpty(opts.Branch)
-                ? GetRemoteDefaultBranch(opts.Repository, workingFolder)
-                : opts.Branch;
             if (!Directory.Exists(workingFolder))
             {
-                Directory.CreateDirectory(workingFolder);
-                Exec("git", $"init", cwd: workingFolder);
-                Exec("git", $"remote add origin {TestDataRepositoryUrl}", cwd: workingFolder);
-                Exec("git", $"{s_gitCmdAuth} fetch origin --progress template", cwd: workingFolder, secrets: s_gitCmdAuth);
+                try
+                {
+                    Directory.CreateDirectory(workingFolder);
+                    Exec("git", $"init", cwd: workingFolder);
+                    Exec("git", $"remote add origin {TestDataRepositoryUrl}", cwd: workingFolder);
+                    Exec("git", $"{s_gitCmdAuth} fetch origin --progress template", cwd: workingFolder, secrets: s_gitCmdAuth);
+                }
+                catch
+                {
+                    if (retryCount++ < 3)
```

Comment:
How about setting `retryCount` as `3` and `if (retryCount-- == 0)`?
And also, can we use polly here to deal with specific exception?

Reply:
fixed